themes/perun/perun/disco-tpl.php

```diff
@@ -23,165 +24,149 @@
 $this->data['head'] .= '<link rel="stylesheet" media="screen" type="text/css" href="' .
     Module::getModuleUrl('perun/res/css/disco.css') . '" />';
 
-$this->data['head'] .= '<script type="text/javascript" src="' .
-    Module::getModuleUrl('perun/res/js/jquery.livesearch.js') . '"></script>';
-
-$this->data['head'] .= '<script type="text/javascript" src="' .
-    Module::getModuleUrl('discopower/assets/js/suggest.js') . '"></script>';
-
-$this->data['head'] .= Disco::searchScript();
-$this->data['head'] .= Disco::showEntriesScript();
-$this->data['head'] .= Disco::setFocus();
-
-const CONFIG_FILE_NAME = 'module_perun.php';
-
-const ADD_INSTITUTION_URL = 'disco.addInstitution.URL';
-const ADD_INSTITUTION_EMAIL = 'disco.addInstitution.email';
-
-const URN_CESNET_PROXYIDP_IDPENTITYID = 'urn:cesnet:proxyidp:idpentityid:';
-
-const WARNING_TYPE_INFO = 'INFO';
-const WARNING_TYPE_WARNING = 'WARNING';
-const WARNING_TYPE_ERROR = 'ERROR';
-
-$warningIsOn = $this->data['warningIsOn'];
-$warningType = $this->data['warningType'];
-$warningTitle = $this->data['warningTitle'];
-$warningText = $this->data['warningText'];
-
+$warningAttributes = $this->data[Disco::WARNING_ATTRIBUTES];
+$this->includeInlineTranslation('{perun:disco:warning_title}', $warningAttributes->getTitle());
+$this->includeInlineTranslation('{perun:disco:warning_text}', $warningAttributes->getText());
 $authContextClassRef = null;
 $idpEntityId = null;
 
 $config = null;
 
 $addInstitutionUrl = '';
 $addInstitutionEmail = '';
+$translate_module = '';
+$wayfConfig = [];
 
+//LOAD CONFIG
 try {
-    $config = Configuration::getConfig(CONFIG_FILE_NAME);
+    $config = Configuration::getConfig(Disco::CONFIG_FILE_NAME);
 } catch (\Exception $ex) {
-    Logger::warning('perun:disco-tpl: missing or invalid module_perun.php config file');
+    Logger::error('perun:disco-tpl: missing or invalid module_perun.php config file');
+    throw $ex;
 }
 
 if ($config !== null) {
```

Comment:
I don't know, if this if is still required.

Reply:
I think it is. If we cannot load the configuration file, the WAYF cannot be displayed. The loading will probably fail before this line gets executed, but I would keep it just to be sure.